src/Microsoft.HttpRepl/ApiConnection.cs

```diff
@@ -183,5 +188,29 @@ private IEnumerable<string> GetSwaggerSearchPaths()
                 return paths;
             }
         }
+
+        private void WriteVerbose(string s)
```

Comment:
I still don't like this.  Whether it's verbose or not should be settled between the command handler and the logger, and should not be a responsibility of this type; `ApiConnection` should just use LogMessage and LogVerboseMessage equivalents. 

To put it another way, `ApiConnection` should know what is normal vs. verbose output, but needn't have any logic or switching behavior about verbosity.  Under my proposed design, the command handler (e.g. BaseHttpCommand, ConnectCommand, HelpCommand, ListCommand, etc) should configure the logger (or IWriteable now) as to whether to show verbose output, then invoke ApiCommand.  

Two hypothetical examples (snippet borrowed from ConnectCommand):
```
bool logVerboseMessages = isVerbosityEnabled || !string.IsNullOrWhiteSpace(swaggerAddress);
using (shellState.ConsoleManager.ShowVerboseMessages(logVerboseMessages))
{
    ApiConnection apiConnection = new ApiConnection(preferences, shellState.ConsoleManager);
    if (!string.IsNullOrWhiteSpace(rootAddress))	
    { ... }
}

// or...
bool logVerboseMessages = isVerbosityEnabled || !string.IsNullOrWhiteSpace(swaggerAddress);
var loggerWithVerbosity = shellState.ConsoleManager.GetWriteable(verbosity: logVerboseMessages);
ApiConnection apiConnection = new ApiConnection(preferences, loggerWithVerbosity);
```

You've mentioned that this area needs a deeper look, so I'm fine with punting this to a later issue/PR to refactor it around.  But let's have a conversation at some point to see what this might look like.

Reply:
Your first example with the using statement is fairly close in concept to what I had been prototyping for the overall system. The only real big difference is that it needs to be in another layer. ConsoleManager is just one possible output location for the "logger", whatever it may be called. File output is another (that will be more important with the upcoming scripting focus) and, potentially, "variables" is a third (mostly for output of http commands, but not necessarily limited to that). Its not clear yet how much of that will be (or even should be) the same logger, or needs for certain to have the same verbosity requirements or anything like that. But the console manager just being one output location seems correct.

So that was where the VerbosityLogger came in originally - it was sort of that intermediate component, but massively stripped down to near uselessness for this purpose... including not having a good creation story because the rest of the plumbing wasn't there to have it available. Thus the awkward creation in a couple places. That was a bad idea on my part, likely just because I had so much of that other design in my head.

So on this second cut, I just went barebones. ApiConnection knows about the switching right now, but in as few places as possible. In the end, we'll probably be changing the constructor to drop the boolean and take an IWhateverLogger instead of an IWritable and change the calls to .Write/.WriteLine to be on that object instead.

Dfinitely want to have a design discussion around this once I get more clarification on some of the things that are in the air for after v5 - scripting and output to files (and anything else that goes along with making it usable for more than just ad hoc testing) are likely high on the priority list and fit into that discussion.